enterprise/api/v1/views.py

```diff
@@ -525,6 +526,39 @@ def unlink_users(self, request, pk=None):  # pylint: disable=unused-argument
 
         return Response(status=HTTP_200_OK)
 
+    @action(detail=False)
+    def algolia_key(self, request, *args, **kwargs):
+        """
+        Returns an Algolia API key that is secured to only allow searching for
+        objects associated with enterprise customers that the user is linked to.
+        """
+
+        if not settings.ENTERPRISE_ALGOLIA_SEARCH_API_KEY:
+            return Response("Algolia API key not configured", status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        queryset = self.queryset.filter(
+            **{
+                self.USER_ID_FILTER: request.user.id,
+                "enterprise_customer_users__linked": True
+            }
+        ).values_list("uuid", flat=True)
+
+        if len(queryset) == 0:
+            raise NotFound("User is not linked to any enterprise customers.")
+
+        secured_key = SearchClient.generate_secured_api_key(
+            settings.ENTERPRISE_ALGOLIA_SEARCH_API_KEY,
+            {
+                "filters": " OR ".join(
+                    f"enterprise_customer_uuids:{str(enterprise_customer_uuid)}"
```

Comment:
```suggestion
                    f"enterprise_customer_uuids:{enterprise_customer_uuid}"
```

Tiny nit: f-strings use `str()` implicitly.